quilt/README.md

```diff
@@ -0,0 +1,241 @@
+[`ray-runner.spec`](ray-runner.spec). `ray.spec` provides abstractions for
+creating ray containers - the function `ray.New` creates `nWorker` worker nodes,
+and one head node, all prefixed with `prefix`.
+
+`ray-runner.spec` defines the infrastructure When you run `ray-runner.spec`, Quilt will
+
+## Step 1: Installing Quilt
+
+### Install Go
+Quilt supports Go version 1.5 or later.
+
+Install Go using your package manager or on the 
+[Golang website](https://golang.org/doc/install).
+
+#### Set up GOPATH
+We recommend reading the overview to Go workplaces 
+[here](https://golang.org/doc/code.html).
+
+Before installing Quilt, you'll need to set up your `GOPATH`. Assuming the root
+of your Go workspace will be `~/gowork`, execute the following `export`
+commands in your terminal to set up your `GOPATH`:
+
+```bash
+export GOPATH=~/gowork
+export PATH=$PATH:$GOPATH/bin
+```
+
+It would be a good idea to add these commands to your `.bashrc` (or
+`.bash_profile` on OS X) so that they do not have to be run again.
+
+### Download and Install Quilt
+Clone the repository into your Go workplace: `go get github.com/NetSys/quilt`.
+
+This command also automatically installs Quilt. If the installation was
+successful, then the `quilt` command should execute in your shell.
+
+### QUILT_PATH
+Your `QUILT_PATH` will be where Quilt looks for imported specs, and where
+specs you download with `quilt get` get placed. You can set this to be anywhere,
+but by default, your `QUILT_PATH` is `~/.quilt`. To set a custom `QUILT_PATH`,
+follow the instructions
+[here](https://github.com/NetSys/quilt/blob/master/docs/Stitch.md#quilt_path).
+
+### Download Imported Specs
+If you look at [`ray.spec`](ray.spec), the first line imports 
+`github.com/NetSys/quilt/specs/stdlib/strings`. To resolve these imports, simply
+execute `quilt get github.com/amplab/ray`. This will clone the ray repository
+into your `QUILT_PATH`, and will also download imports of the specs that were
+downloaded.
+
+## Step 2: Configure `ray-runner.spec` & Cloud Provider
+In [`ray-runner.spec`](ray-runner.spec), there are two things that will have to
+be filled in.
+
+### Configure SSH Authentication
+Quilt-managed Machines use public key authentication to control SSH access.
+To start using the Ray cluster, we will need to access the Master VM.
+
+If you would like to use `githubKey` authentication, go to line 24 of 
+[`ray-runner.spec`](ray-runner.spec) and fill in  `(define githubKey
+"YOUR_GITHUB_USERNAME")` appropriately. Instructions on adding a key to your
+GitHub account can be found
+[here](https://help.github.com/articles/generating-an-ssh-key/).
+
+If you would ike to use an SSH key instead, replace `(define githubKey
+"YOUR_GITHUB_USERNAME")` with `(define sshkey "YOUR_PUBLIC_KEY")`.
+
+`"YOUR_PUBLIC_KEY"` should be a string with the contents of your
+`~/.ssh/id_rsa.pub`, e.g.: `(define sshkey "ssh-rsa
+AAAAB3NzaC1yc2EAAAADAQABAAABAQ example@example.com")`.
+
+### Choose a Namespace
```

Comment:
Why not just run in the default namespace?  Then we can simplify the docs further.